cpp/src/sort/segmented_sort.cu

```diff
@@ -34,6 +39,195 @@ namespace {
  */
 enum class sort_method { STABLE, UNSTABLE };
 
+/**
+ * @brief Functor performs faster sort on eligible columns
+ */
+struct column_fast_sort_fn {
+  /**
+   * @brief Run-time check for faster sort on an eligible column
+   */
+  static bool is_fast_sort_supported(column_view const& col)
+  {
+    return !col.has_nulls() and cudf::is_numeric(col.type()) and
```

Comment:
I'm a little unsettled that we are using `is_numeric and is_floating_point` at runtime, but `is_integral` at compile time. Can we add a trait `cudf::is_integral(data_type)` that acts like `std::is_integral`, to align with how we have `cudf::is_floating_point`?

Note that `cuda::std::is_integral` will return true for int128, but `std::is_integral` will not.

More broadly, we need to decide if we want to handle decimals using this same code path. In theory that should be fine, right? What about other types using integral representations, like timestamps or durations? Should they get the fast sort? To my knowledge there is no reason (or special representation for those types) that would prohibit the fast sort.

Because we are now able to define this in a single location, we should ensure that we cover all the appropriate types with the fastest method available.

Reply:
Unfortunately, CUB `DeviceSegmentedSort` does not seem to like `timestamp` and `duration` types. 
Also it does not seem to like `int128` either so this means the current implementation will fail for DECIMAL128 columns.

Reply:
Is it well-defined to sort the `timestamp` or `duration` types "as if" they're integral, e.g. with a type cast?

Totally fine for this skip `int128` -- but should still be viable for 32 and 64 bit decimals, I'd think.

Reply:
> Is it well-defined to sort the `timestamp` or `duration` types "as if" they're integral, e.g. with a type cast? (I've asked this before and can't quite recall the answer.)

I don't that is allowed. Maybe I don't understand what you are asking. There is no equivalent dispatch-storage-type for the chrono types. The chrono types have their own comparators and perhaps that is why `DeviceSegmentedSort` cannot handle them.

Reply:
Okay, thanks. I always forget that they have their own comparators.